modules/fhir-structure/test/blaze/fhir/spec_test.clj

```diff
@@ -58,13 +59,45 @@
       {:resourceType "Patient"
        :id "0"}))
 
+
   (testing "invalid resources"
     (are [resource] (not (fhir-spec/valid? resource))
+      {}
       {:resourceType "Patient"
        :id ""}
       {:resourceType "Patient"
        :id "/"})))
 
+(deftest explain-data
+  (testing "valid resources"
+    (are [resource] (not (fhir-spec/explain-data resource))
+      {:resourceType "Patient" :id "."}
+      {:resourceType "Patient" :id "0"}))
+
+
+  (testing "empty resource"
+    (given (fhir-spec/explain-data {})
+      [:fhir/issues 0 :fhir.issues/severity] := "error"
+      [:fhir/issues 0 :fhir.issues/code] := "value"
+      [:fhir/issues 0 :fhir.issues/diagnostics] :=
+      "Given resource does not contain :resourceType key!"))
+
+
+  (testing "invalid resource"
+    (given (fhir-spec/explain-data {:resourceType "Patient"
+                                    :name [{:use "" :text []}]})
+      [:fhir/issues 0 :fhir.issues/severity] := "error"
+      [:fhir/issues 0 :fhir.issues/code] := "invariant"
+      [:fhir/issues 0 :fhir.issues/diagnostics] :=
+      "Error on Value ``. Expected type is `code`,
+      regex `[^\\s]+(\\s[^\\s]+)*`."
+      [:fhir/issues 0 :fhir.issues/expression] := "name[0].use"
+      [:fhir/issues 1 :fhir.issues/severity] := "error"
+      [:fhir/issues 1 :fhir.issues/code] := "invariant"
+      [:fhir/issues 1 :fhir.issues/diagnostics] :=
+      "Error on Value `[]`. Expected type is `string`."
+      [:fhir/issues 1 :fhir.issues/expression] := "name[0].text")))
```

Comment:
please add tests of invalid backbone elements and non-primitive elements like HumanName or Quantity

Reply:
Done